### PR TITLE
Use formatter in docker port

### PR DESCRIPTION
**Pull request by boaz0**
**- What I did**
Used formatter to print port list
related to #30431

**- How I did it**
* Add cli/command/formatter/port.go
* Replace fmt.Println with formatter.PortWrite

**- How to verify it**
```bash
$ TESTFLAGS='-check.f DockerSuite.TestPort*' hack/make.sh test-integration-cli
$ TESTFLAGS='-check.f DockerSuite.TestUnpublishedPort*' hack/make.sh test-integration-cli
```

### CODE REVIEW

File: cli/command/formatter/port.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+				return err
+			}
+		}
+		return nil
+	}
+	subCtx := &portContext{}
+	return ctx.Write(subCtx, render)
+}
+
+type portContext struct {
```
Comment (AkihiroSuda) [inline cli/command/formatter/port.go]: MarshalJSON please

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

File: cli/command/formatter/port.go (hunk 1 above)
Comment (ripcurld0) [inline cli/command/formatter/port.go]: @AkihiroSuda thanks for the feedback.
I will add it to the commit but just a small question: from where MarshalJSON is called?
Out of curiosity I tried to find where it's used but failed.
If you can give me some information why we are using it, I will be grateful.

Thanks 😃

Comment (AkihiroSuda) [inline cli/command/formatter/port.go]: for `--format {{json .}}`

**Comment by thaJeztah** [comment]
Let me be "that one"; I wonder if adding `--format` to `docker port` is a bit feature creep (I know some maintainers actually thought of deprecating the command, given that `docker inspect` also allows obtaining that information).

Are there some practical examples of using `--format` with this command (other than "for consistency"?)

**Comment by ripcurld0** [comment]
No, intentionally this PR was meant for consistency unless @AkihiroSuda had other ideas.
🐕

**Comment by AkihiroSuda** [comment]
I'm ok to deprecate `docker port` (then this PR is unlikely to get merged, sorry and thank you for working on this @ripcurld0 :sweat_smile: )

**Comment by thaJeztah** [comment]
Thanks @ripcurld0! Let me slightly clarify; there's no "active" effort to remove/deprecate `docker port`, but keeping it simple as-is, and not overload with new features is good for now. If actual use-cases come up at some point, we can revisit, but for now, let's not touch it :smile:

**Comment by ripcurld0** [comment]
@thaJeztah I can't agree with you more. The last thing I want is to push code that's not necessary for this awesome project❗️ 😹

**Comment by AkihiroSuda** [comment]
@thaJeztah 

Do we have an GitHub issue to discuss deprecation of `docker port`?

FYI:
- first commit with `docker port` (Mar 6 2013): https://github.com/docker/docker/commit/410ebe
- first commit with `docker inspect` (Mar 11 2013): https://github.com/docker/docker/commit/06d420d2ead65cb5af760d7946d8844f568f4d65


There has been both since really early days

**Comment by thaJeztah** [comment]
@AkihiroSuda no, I don't think there's an issue to discuss it. I know it has come up a few times (maintainers don't liking the command), but it also came up that (because it's around for so long), it's likely a command that we just have to "deal with" being there

**Pull request closed without merging**